### DISCUSSION TITLE
Space as leader key?

### DISCUSSION THREAD

**Discussion by azzamsa** [wezterm/wezterm#2252] [Q&A] [answered]
Hi.

Is it possible to use `space` as leader key?
I want to mimic Doom Emacs key into wezterm. Something like:
- `space + w + v`
- `space + w + w`

If I can use `space`, is the multiple key sequence such above supported?

Thanks for wezterm.

**Comment by wez** [comment, marked as answer]
You can specify any key that registers as a KeyEvent when `debug_key_events=true` as the leader.

Instead of leader keys, I'd suggest that you take a look at [Key Tables](https://wezfurlong.org/wezterm/config/key-tables.html) for a way to build stacked/layered key assignments for multi-stage key presses.  eg: have space activate the first key table, and in that key table have `w` activate a second key table which has your final 'v' and 'w' assignments.

**Reply by azzamsa** [reply]
If `one_shot` and `replace_current` is optional. Why excluding them always raises an error.

![image](https://user-images.githubusercontent.com/17734314/178643706-c06acc68-56f9-4f69-a3e9-29afa33368a2.png)

**Reply by MuhammedZakir** [reply]
Have you tried latest nightly? It should already be fixed.

ref: https://github.com/wez/wezterm/issues/2179

**Reply by azzamsa** [reply]
Solved after upgrade to nightly. Thanks.

**Reply by azzamsa** [reply]
Key table is amazing. It is rarely found in terminal or even daily apps. Thanks a lot for Wezterm!

**Comment by eugenesvk** [comment]
How do you avoid interference with typing the actual `` space?
